src/hotspot/share/oops/method.cpp

```diff
@@ -566,16 +566,23 @@ MethodCounters* Method::build_method_counters(Thread* current, Method* m) {
   methodHandle mh(current, m);
   MethodCounters* counters;
   if (current->is_Java_thread()) {
-    // For when TRAPS is JavaThread.
-    counters = MethodCounters::allocate(mh, current->as_Java_thread());
+    Thread* THREAD = current;
+    counters = MethodCounters::allocate(mh, THREAD);
+    if (HAS_PENDING_EXCEPTION) {
+      CLEAR_PENDING_EXCEPTION;  // MethodData above doesn't clear exception
```

Comment:
I don't understand the comment.

Reply:
It's sort of a question, because the above code that does the same thing for MethodData doesn't clear the exception.  In this case we should clear the exception I believe.